code/game/machinery/vending/marine_vending.dm

```diff
@@ -233,13 +235,125 @@
+	icon_state = "lascharger"
+	icon_vend = "lascharger_dispensing"
+	icon_deny = "lascharger_denied"
+	req_access = null
+	req_access_txt = "0"
+	req_one_access = null
+	req_one_access_txt = "9;2;21"
+	wrenchable = TRUE
+	drag_delay = 0
+	anchored = 0
+	idle_power_usage = 1
+	vend_power_usage = 50
+	machine_current_charge = 20000 //integrated battery for recharging energy weapons. Normally 10000.
+	machine_max_charge = 20000
+
+	product_ads = "Lasgun running low? Recharge here!;Need a charge?;Power up!;Electrifying!;Empower yourself!"
+	products = list(
+					/obj/item/ammo_magazine/lasgun/M43 = 10,
+					/obj/item/ammo_magazine/lasgun/M43/highcap = 2,
+					)
+
+	contraband =   list()
+
+	premium = list()
 
+	prices = list()
+
+/obj/machinery/vending/lasgun/New()
+	..()
+	update_icon()
+
+/obj/machinery/vending/lasgun/update_icon()
+	if(machine_current_charge <= 0)
+		icon_state = "lascharger-off"
+	else if(machine_current_charge > round(machine_max_charge*0.75))
+		icon_state = "lascharger"
+	else if(machine_current_charge > round(machine_max_charge*0.5))
+		icon_state = "lascharger_75"
+	else if(machine_current_charge > round(machine_max_charge*0.25))
+		icon_state = "lascharger_50"
+	else
+		icon_state = "lascharger_25"
+
+/obj/machinery/vending/lasgun/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/ammo_magazine/lasgun))
+		stock(W, user, TRUE) //
+		//to_chat(user, "<span class='warning'>DEBUG: Lasgun vendor restocked with [W].</span>")
+		return TRUE
+	. = ..()
+
+/obj/machinery/vending/lasgun/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 0)
+
+	var/list/display_list = list()
+	var/list/display_records = list()
+	display_records += product_records
+	if(extended_inventory)
+		display_records += hidden_records
+	if(coin)
+		display_records += coin_records
+	for (var/datum/data/vending_product/R in display_records)
+		var/prodname = adminscrub(R.product_name)
+		if(R.amount) prodname += ": [R.amount]"
+		else prodname += ": SOLD OUT"
+		if(R.price) prodname += " (Price: [R.price])"
+		prodname = "<color = [R.display_color]>[prodname]</color>"
+		display_list += list(list("product_name" = prodname, "product_color" = R.display_color, "amount" = R.amount, "prod_index" = GetProductIndex(R), "prod_cat" = R.category))
+
+	var/list/data = list(
+		"vendor_name" = name,
+		"currently_vending_name" = currently_vending ? sanitize(currently_vending.product_name) : null,
+		"premium_length" = premium.len,
+		"ewallet" = ewallet ? ewallet.name : null,
+		"ewallet_worth" = ewallet ? ewallet.worth : null,
+		"coin" = coin ? coin.name : null,
+		"displayed_records" = display_list,
+	)
+
+	data["energy"] = machine_current_charge
+	data["maxEnergy"] = machine_max_charge
+
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+
+	if (!ui)
+		ui = new(user, src, ui_key, "vending_machine.tmpl", name , 450, 600)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
```

Comment:
This seems largely a copypaste of `/obj/machinery/vending/ui_interact()`. You can simply use `..()` if you just want to add two entries to `data`.